Kernel/Syscalls/open.cpp

```diff
@@ -74,6 +73,7 @@ ErrorOr<FlatPtr> Process::sys$open(Userspace<Syscall::SC_open_params const*> use
 
     return m_fds.with_exclusive([&](auto& fds) -> ErrorOr<FlatPtr> {
         u32 fd_flags = (options & O_CLOEXEC) ? FD_CLOEXEC : 0;
+        auto fd_allocation = TRY(allocate_fd());
```

Comment:
Judging by 5331d243c690c70e431e2f8d260eacab19946c2b you can just do `fds.allocate()` if you already have the file descriptor table locked anyways.